staging/src/k8s.io/client-go/tools/cache/shared_informer.go

```diff
@@ -130,11 +131,24 @@ import (
 // A delete notification exposes the last locally known non-absent
 // state, except that its ResourceVersion is replaced with a
 // ResourceVersion in which the object is actually absent.
+//
+// The informational methods (EventHandlerCount, IsStopped, IsStarted)
+// are intended to be used to manage informers in any upper informer
+// management layer for creating and destroying informers on-the fly when
+// adding or removing handlers.
+// Beware of race conditions: Such a layer must hide the basic informer
```

Comment:
This is too big of a caveat IMO, we have to choose a different paradigm. Perhaps the higher layer could e.g. register callbacks that can be called by the shared informer at the appropriate times.

Reply:
I've removed those methods again causing the potential for races. `IsStopped` keeps stopped forever once true is reported. Therefore we should be able to keep it.